src/common/transformations/tests/common_optimizations/matmul_multiply_fusion.cpp

```diff
@@ -77,24 +78,71 @@ TEST_F(TransformationTestsF, MatMulMultiplyFusionConstantTransposedWeightsNonSca
+        a.run_passes(model);
+    }
+
+    {
+        auto data = std::make_shared<opset8::Parameter>(element::f32, Shape{2, 3});
+        auto weights = opset8::Constant::create(element::f32, Shape{2, 3}, {1, 1, 1, 1, 1, 1});
+        auto matmul = std::make_shared<opset8::MatMul>(data, weights, false, true);
+        model_ref = std::make_shared<Model>(NodeVector{matmul}, ParameterVector{data});
+    }
+    comparator.enable(FunctionsComparator::CmpValues::CONST_VALUES);
+}
+
+TEST_F(TransformationTestsF, MatMulMultiplyFusionConstantWeightsMarkedToKeepSrcPrecision) {
+    {
+        auto data = std::make_shared<opset8::Parameter>(element::f32, Shape{1, 2, 4, 3});
+        auto weights = opset8::Constant::create(element::f32, Shape{3, 2}, {1, 2, 3, 4, 5, 6});
+        mark_as_dequantization_node(weights);
+        auto matmul = std::make_shared<opset8::MatMul>(data, weights);
+        auto mul_const = opset8::Constant::create(element::f32, Shape{1, 1, 1, 2}, {2, 3});
+        auto mul = std::make_shared<opset8::Multiply>(matmul, mul_const);
+        model = std::make_shared<Model>(NodeVector{mul}, ParameterVector{data});
+        model_ref = model->clone();
+
+        manager.register_pass<ov::pass::MatMulMultiplyFusion>();
+    }
+    comparator.enable(FunctionsComparator::CmpValues::CONST_VALUES);
+}
+
+TEST_F(TransformationTestsF, MatMulMultiplyFusionNonConstant) {
     {
         auto data = std::make_shared<opset8::Parameter>(element::f32, Shape{2, 3});
         auto weights = std::make_shared<opset8::Parameter>(element::f32, Shape{2, 3});
         auto matmul = std::make_shared<opset8::MatMul>(data, weights, false, true);
         auto mul_const = opset8::Constant::create(element::f32, Shape{1, 2}, {4, 5});
         auto mul = std::make_shared<opset8::Multiply>(matmul, mul_const);
         model = std::make_shared<Model>(NodeVector{mul}, ParameterVector{data, weights});
+        model_ref = model->clone();
 
         manager.register_pass<ov::pass::MatMulMultiplyFusion>();
     }
+    comparator.enable(FunctionsComparator::CmpValues::CONST_VALUES);
+}
 
+TEST_F(TransformationTestsF, MatMulMultiplyFusionWithQuantizedWeights) {
     {
-        auto data = std::make_shared<opset8::Parameter>(element::f32, Shape{2, 3});
-        auto weights = std::make_shared<opset8::Parameter>(element::f32, Shape{2, 3});
-        auto mul_const = opset8::Constant::create(element::f32, Shape{2, 1}, {4, 5});
-        auto mul = std::make_shared<opset8::Multiply>(weights, mul_const);
-        auto matmul = std::make_shared<opset8::MatMul>(data, mul, false, true);
-        model_ref = std::make_shared<Model>(NodeVector{matmul}, ParameterVector{data, weights});
+        auto data = std::make_shared<opset8::Parameter>(element::i8, Shape{2, 3});
+        auto weights = opset8::Constant::create(element::i8, Shape{2, 3}, {1, 2, 3, 4, 5, 6});
+        auto matmul = std::make_shared<opset8::MatMul>(data, weights, false, true);
```

Comment:
If we are trying to imitate a graph after LPT, `matmul` output precision must be fp32 (because multiplication of 2 i8 values might be out of i8 range). Can we do that (we need to cover Matmul in TypeRelaxed for that: [example](https://github.com/openvinotoolkit/openvino/blob/master/src/tests/ov_helpers/ov_lpt_models/src/mat_mul.cpp#L236-L242))?